### PR TITLE
Promote CSINodeExpandSecret to GA

### PR DESCRIPTION
**Pull request by humblec**
Upstream PR # https://github.com/kubernetes/kubernetes/pull/121303

Signed-off-by: Humble Chirammal <humble.devassy@gmail.com>

### CODE REVIEW

**Comment by tengqm** [comment]
@humblec Please add a link to the upstream PR so that we know if and when the promotion has landed.
/hold

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -80,6 +80,7 @@ For a reference to old feature gates that are removed, please refer to
 | `CSIMigrationPortworx` | `false` | Beta | 1.25 | |
 | `CSINodeExpandSecret` | `false` | Alpha | 1.25 | 1.26 |
 | `CSINodeExpandSecret` | `true` | Beta | 1.27 | |
```
Comment (Shubham82) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:82]:
```suggestion
| `CSINodeExpandSecret` | `true` | Beta | 1.27 | 1.28 |
```
Mention the release until it is in beta i.e. k8s v1.28

**Comment by Shubham82** [comment]
Here is the upstream PR: https://github.com/kubernetes/kubernetes/pull/121303

please remove the hold label once upstream PR merges.

**Comment by Shubham82** [comment]
I think we have to update the below content for `CSINodeExpandSecret` according to the GA state in the [volumes](https://kubernetes.io/docs/concepts/storage/volumes/) docs:

> If you are running Kubernetes Version 1.25 or 1.26, you must enable the [feature gate](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/) named CSINodeExpandSecret for each kube-apiserver and for the kubelet on every node. In Kubernetes version 1.27 this feature has been enabled by default and no explicit enablement of the feature gate is required. You must also be using a CSI driver that supports or requires secret data during node-initiated storage resize operations.

**Comment by humblec** [comment]
Updated the section too. ptal. thanks !

**Comment by humblec** [comment]
> @humblec Please add a link to the upstream PR so that we know if and when the promotion has landed.
> /hold

Sure, added it in the PR description too. thanks !

File: content/en/docs/concepts/storage/volumes.md [hunk 2]
```diff
@@ -1086,7 +1086,7 @@ persistent volume:
 * If you are running Kubernetes Version 1.25 or 1.26, you must enable
   the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
   named `CSINodeExpandSecret` for each kube-apiserver and for the kubelet on every
-  node. In Kubernetes version 1.27 this feature has been enabled by default
+  node. In Kubernetes version 1.29 this feature has been enabled by default
```
Comment (tengqm) [inline content/en/docs/concepts/storage/volumes.md:1089]: This statement about history is correct and should not be changed.
If you want, you can add that this feature is GA'ed in v1.29.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -79,7 +79,8 @@ For a reference to old feature gates that are removed, please refer to
 | `CSIMigrationPortworx` | `false` | Alpha | 1.23 | 1.24 |
 | `CSIMigrationPortworx` | `false` | Beta | 1.25 | |
 | `CSINodeExpandSecret` | `false` | Alpha | 1.25 | 1.26 |
-| `CSINodeExpandSecret` | `true` | Beta | 1.27 | |
+| `CSINodeExpandSecret` | `true` | Beta | 1.27 | 1.28 |
+| `CSINodeExpandSecret` | `true` | GA   | 1.29 | |
```
Comment (tengqm) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:83]: GA'ed feature should be moved to the second table.

File: content/en/docs/concepts/storage/volumes.md (hunk 2 above)
Comment (humblec) [reply content/en/docs/concepts/storage/volumes.md]: Better I kept this unchanged.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 3 above)
Comment (humblec) [reply content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Sure.. thanks! @tengqm

**Comment by tengqm** [comment]
@humblec Need your help to resolve the conflicts.

**Comment by humblec** [comment]
> @humblec Need your help to resolve the conflicts.

Done @tengqm .. Thanks!

**Comment by tengqm** [comment]
/lgtm
/approve
/hold
Remove hold when upstream PR is merged.

**Comment by Shubham82** [comment]
remove the hold label as upstream PR (https://github.com/kubernetes/kubernetes/pull/121303) is merged now.
/unhold

**Pull request merged**